### DISCUSSION TITLE
CIR  for multiple UEs

### DISCUSSION THREAD

**Discussion by Advaith2212** [NVlabs/sionna#239] [Q&A] [answered]
I have some doubt in generating `CIR` for multiple (**e.g. 1000**)UEs. I followed the following procedure:
  1. Generated a coverage map and using it to sampled positions using `cm.sample_positions()` 
  2. Since no of UE is not so huge, I used the positions to deploy `rx antennas`
  3. Used the `scene.compute_paths()` to generate `a` and `tau`
  4. Shape of **a** : `1,num_rx,num_rx_ant,num_tx,num_tx_ant,num_paths`
  5. When I visualize the CIR between a particular UE and BS, I am seeing negative `delay` in some paths.
 
![Screenshot from 2023-10-19 15-30-57](https://github.com/NVlabs/sionna/assets/64831500/832d9e24-75b4-4e1e-819c-bace6b119ff8)

![Screenshot from 2023-10-19 15-31-01](https://github.com/NVlabs/sionna/assets/64831500/419fabf5-1518-4612-92e0-ce7f415a9b64)

  Can you let me know if my procedure to generate CIR correct ?

**Comment by jhoydis** [comment]
That's all correct. The paths with negative delay have also zero amplitude and will be ignored.

**Reply by Advaith2212** [reply]
In the `t` uploaded above, the third path has a negative delay. What does it imply ?

**Reply by jhoydis** [reply, marked as answer]
It means that this is a non-existent path. If you check the corresponding amplitude, it will be equal to zero.
The number of paths per transmit-receive antenna pair is not necessarily equal, but we need to return a tensor where each CIR has the same number of paths. In order to indicate invalid (= non existent) paths, we have chosen to represent them with a negative delay and zero amplitude.
